### PR TITLE
Batch commands no longer throw error upon precondition failures.

### PR DESCRIPTION
**Pull request by williexu**
---
Closes: https://github.com/Azure/azure-cli/issues/5286
-batch commands should not fail upon preconditions set by user

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5702
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py [hunk 1]
```diff
@@ -207,3 +207,16 @@ def normalize_blob_file_path(path, name):
     if path:
         name = path_sep.join((path, name))
     return path_sep.join(os.path.normpath(name).split(os.path.sep)).strip(path_sep)
+
+
+def check_precondition_success(func):
+    def new_func(*args, **kwargs):
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]: `wrapper`

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py [hunk 2]
```diff
@@ -207,3 +207,16 @@ def normalize_blob_file_path(path, name):
...
+        from azure.common import AzureHttpError
+        try:
+            return (True, func(*args, **kwargs))
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]: You don't need the `()`

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py [hunk 3]
```diff
@@ -207,3 +207,16 @@ def normalize_blob_file_path(path, name):
...
+        except AzureHttpError as ex:
+            # Precondition failed error
+            if ex.status_code != 412:
+                raise ex
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]: `raise`

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py [hunk 4]
```diff
@@ -207,3 +207,16 @@ def normalize_blob_file_path(path, name):
...
+            return (False, None)
```
Comment (troydai) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]: no need `()`

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py (hunk 2 above)
Comment (troydai) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]:
```
return 1, 2  # => (1,2)
return 1 => 1
```

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py (hunk 3 above)
Comment (troydai) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py]: Also, can we log a warning here?

**Comment by troydai** [comment]
Looks good besides the comments.

**Comment by williexu** [comment]
@troydai could you review my feedback changes?

**Pull request merged**